### ISSUE TITLE
Thumbnail Preview keep proportion

### ISSUE THREAD

**Issue by WWebber** [LycheeOrg/Lychee-v3#333]
The thumbnail preview fits the image independently of the image width & height into the thumbnail size. Is there any option where I can keep the aspect ratio resp. use the same width/height scaling? Some of the images get stretched horizontal, some of them get even truncated. So what's the trick?

**Comment by WWebber** [comment]
Ok. After investigating the source code, I'm going to fix it myself for my needs..

**Comment by ildyria** [comment]
I'm sorry I did not really get what you wanted because I get the feeling this is already implemented. Have you had a look at the Justified Layout (Flickr style) which is already available ?

I would also highly advise you to have a look at https://github.com/LycheeOrg/Lychee-laravel which could be considered at the mature Beta of the v4.
This repository is quite stale as you can see.

**Comment by WWebber** [comment]
As far as I understood the Laravel version requires more than just a simple domain server from my provider? I don't have an SSH terminal for setup..
BTW: what's the justified layout? I would adapt the code in two main aspects: a) make better thumbnails and b) change compression & resolution of uploaded images. c) I also would like to change the default thumbnail size.

**Comment by ildyria** [comment]
> As far as I understood the Laravel version requires more than just a simple domain server from my provider? I don't have an SSH terminal for setup..

I actually we are trying to solve this ssh requirement.
With https://github.com/LycheeOrg/Lychee-Laravel/pull/361 we should now have an installer which will check the requirements. When we produce a release version, we also include the vendor folders so that composer is not needed for the install. This Pull request should allow you to also apply the migrations and get started. :)

> BTW: what's the justified layout? 

That:
![2020-02-08_1920x1080_12:32:30](https://user-images.githubusercontent.com/627094/74084443-80be4580-4a6f-11ea-976f-776be63f3a83.png)

> a) make better thumbnails and b) change compression & resolution of uploaded images. c) I also would like to change the default thumbnail size.

a) I don't get what you mean by better thumbnails, this is quite vague.  
b) is already available
c) if you are not talking about the Thumbnail but "small" and "medium" sizes this is also already available in the advanced settings (the RED "more" button at the bottom of the settings page).
See:
![2020-02-08_1920x1080_12:39:28](https://user-images.githubusercontent.com/627094/74084502-0e9a3080-4a70-11ea-87fe-624582d938e6.png)

**Comment by WWebber** [comment]
Thanks for the reply. And I read already the thread "LycheeOrg/Lychee-Laravel#361", but honestly I did not understand how I shall now install what how? A simple tutorial would be helpful for 1) "migration from Lychee Std" and 2) starting from scratch.

"we should now have an installer": where is it and can I test it?

a) I mainly meant to control how the thumbnails will be created (for the "albums" view and inside the album itself). The preview image you posted goes in the right direction, if it will be customizeable. I'd like to get N by M pixel thumbnails, showing simply the scaled image, with out any cropping.
b) ok
c) works only in the Laravel version?

ohh.. additionally my provider has only PHP 7.2 available..

**Comment by d7415** [comment]
> A simple tutorial would be helpful for 1) "migration from Lychee Std" and 2) starting from scratch.

We have https://github.com/LycheeOrg/Lychee-Laravel/wiki/Install which is linked from the README

> "we should now have an installer": where is it and can I test it?

It was merged into master today. LycheeOrg/Lychee-Laravel/pull/361

> ohh.. additionally my provider has only PHP 7.2 available..

We only removed support today (as it's EOL) but we won't intentionally break it. Admittedly, that's not a guarantee of future support.

**Comment by ildyria** [comment]
> Thanks for the reply. And I read already the thread "[LycheeOrg/Lychee-Laravel#361](https://github.com/LycheeOrg/Lychee-Laravel/pull/361)", but honestly I did not understand how I shall now install what how? A simple tutorial would be helpful for 1) "migration from Lychee Std" and 2) starting from scratch.

2) We currently do not have a "release" version available yet with the installer (very soon).
I am thinking about waiting for the merge of https://github.com/LycheeOrg/Lychee-Laravel/pull/430 before generating the next release.

Now that we have an installer, I'm seriously considering removing the `beta` part of Lychee-Laravel and finally have it officially as current. :)

> "we should now have an installer": where is it and can I test it?

Once the release is done, you can just extract it. and then accessing the install.php file from your web browser it will guide you through the install. 

> a) I mainly meant to control how the thumbnails will be created (for the "albums" view and inside the album itself). The preview image you posted goes in the right direction, if it will be customizeable. I'd like to get N by M pixel thumbnails, showing simply the scaled image, with out any cropping.
> b) ok
> c) works only in the Laravel version? 

a) N by M exactly is not guaranteed because of proportion, but there is no crop in the justified layout.  
c) It is also available in version 3 since 3.2.2 and 3.2.4

> ohh.. additionally my provider has only PHP 7.2 available..

You will just need to edit a small file and everything will work for 7.2.

**Takes note to add this into the FAQ**

**Comment by WWebber** [comment]
Ok. I'm patient and will postbone my software activities.. Thx so far..

I gave 3.2.16 a test. In this version I cannot organize the order of images by hand anymore - i.e. by drag/drop?! (usually only commercial software loses always already well introduced and well working options for making extra money with service packs)

Concerning the N by M thumbnail size: I would create a black empty square image and fit the original image centered into.

**Comment by d7415** [comment]
v4 has now been released and is available at https://github.com/LycheeOrg/Lychee

> I gave 3.2.16 a test. In this version I cannot organize the order of images by hand anymore - i.e. by drag/drop?! (usually only commercial software loses always already well introduced and well working options for making extra money with service packs)

As far as I can remember, that was never a feature. I vaguely remember hearing that some features may have been added to electerious's version after the LycheeOrg fork, but I can't say for sure. For v4 this is being tracked [here](https://github.com/LycheeOrg/Lychee/issues/58). I don't anticipate it being backported to v3.

> Concerning the N by M thumbnail size: I would create a black empty square image and fit the original image centered into.

This part is `wontfix` for v3. If you would like it that way in v4, feel free to open an issue there.
